posthog/temporal/batch_exports/monitoring.py

```diff
@@ -0,0 +1,220 @@
+import datetime as dt
+import json
+from dataclasses import dataclass
+from uuid import UUID
+
+from temporalio import activity, workflow
+from temporalio.common import RetryPolicy
+
+from posthog.batch_exports.service import (
+    aget_active_event_batch_exports,
+    aupdate_expected_records_count,
+)
+from posthog.batch_exports.sql import EVENT_COUNT_BY_INTERVAL
+from posthog.temporal.batch_exports.base import PostHogWorkflow
+from posthog.temporal.common.clickhouse import get_client
+
+
+class NoActiveBatchExportsFoundError(Exception):
+    """Exception raised when no active events batch exports are found for a given team."""
+
+    def __init__(self, team_id: int):
+        super().__init__(f"No active events batch exports found for team {team_id}")
+
+
+class NoValidBatchExportsFoundError(Exception):
+    """Exception raised when no valid events batch exports are found for a given team."""
+
+    def __init__(self, message: str = "No valid events batch exports found for team"):
+        super().__init__(message)
+
+
+@dataclass
+class BatchExportMonitoringInputs:
+    """Inputs for the BatchExportMonitoringWorkflow.
+
+    Attributes:
+        team_id: The team id to monitor batch exports for.
+    """
+
+    # TODO - make this a list of team ids or single?
+    # or maybe a (list of) batch export id(s) instead?
```

Comment:
maybe easier to use a batch export id instead in case the team has more than one set up?

Reply:
I think it would be better to take a `batch_export_id` :thinking: But also fine with trying `team_id` and see how that works in practice if you are not convinced yet (I am not, maybe the solution is both...)

Reply:
maybe I'll change this to take a `batch_export_id` for now since it makes some of the logic simpler and then can extend this to take a `team_id` if useful in future

Reply:
Sounds good :+1: Thank you.